recipes/b2/portable/test_package/conanfile.py

```diff
@@ -1,9 +1,12 @@
-from conans import ConanFile, tools
-import os
+from conan import ConanFile
 
 
-class TestPackgeConan(ConanFile):
+class TestPackageConan(ConanFile):
     settings = "os", "arch"
+    test_type = "explicit"
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
 
     def test(self):
         self.run("b2 -v", run_environment=True)
```

Comment:
The key think here it to make sure that the 1.x and 2.0 both work correctly with the new `VirtualBuildEnv1
So youll notice that was added. Also `run_enviroment` was removed because thats not supported in 2.0


```suggestion
from conan import ConanFile


class TestPackageConan(ConanFile):
    settings = "os", "arch", "compiler", "build_type"
    generators = "VirtualBuildEnv"
    test_type = "explicit"

    def build_requirements(self):
        self.tool_requires(self.tested_reference_str)

    def test(self):
        self.run("b2 -v")
```

Reply:
The test_package setting's are not used to determine configuration. They are just copy paste from the template. You are absolutely correct they are not required since they are deleted from the package. It's one of those it's just easier to copy paste

They have no effect with how the client works so they are safe to leave generically.

***

It's murdered!

![image](https://user-images.githubusercontent.com/16867443/199591940-f8a887a5-9267-47e2-adef-0ce9235feda1.png)

Which transition guide are you reference to? I'll make sure to go add it

Reply:
@prince-chrismc so, it wasn't properly working in `develop2`, but it was a still allowed to be used (for some reason)? shouldn't it be forbidden in v2 altogether to avoid confusion?